app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/TimelineViewModel.kt

```diff
@@ -271,6 +272,11 @@ abstract class TimelineViewModel(
             is PreferenceChangedEvent -> {
                 onPreferenceChanged(event.preferenceKey)
             }
+            is FilterUpdatedEvent -> {
+                if (filterContextMatchesKind(kind, event.filterContext)) {
+                    fullReload()
```

Comment:
`invalidate()` isn't enough here because the application of the filters occurs on the server